test/extended/registry/registry.go

```diff
@@ -0,0 +1,155 @@
+
+		g.By("restoring manifest...")
+		image, err = oc.AsAdmin().Client().Images().Get(imageDigest)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		image.DockerImageManifest = string(manifest)
+
+		_, err = oc.AsAdmin().Client().Images().Update(image)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("checking that the manifest is present in the image...")
+		image, err = oc.AsAdmin().Client().Images().Get(imageDigest)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(len(image.DockerImageManifest)).Should(o.BeNumerically(">", 0))
+		o.Expect(image.DockerImageManifest).Should(o.Equal(string(manifest)))
+
+		g.By("getting image manifest from docker-registry one more time...")
+		_, manifest, err = conn.ImageManifest(oc.Namespace(), repoName, tagName)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(len(manifest)).Should(o.BeNumerically(">", 0))
+
+		g.By("waiting until image is updated...")
+		err = waitForImageUpdate(oc, image)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("checking that the manifest was removed from the image...")
+		image, err = oc.AsAdmin().Client().Images().Get(imageDigest)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(len(image.DockerImageManifest)).Should(o.Equal(0))
+
+		g.By("getting image manifest from docker-registry to check if he's available...")
+		_, manifest, err = conn.ImageManifest(oc.Namespace(), repoName, tagName)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(len(manifest)).Should(o.BeNumerically(">", 0))
+
+		g.By("pulling image...")
+		authCfg, err := exutil.BuildAuthConfiguration(registryURL, oc)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		opts := dockerClient.PullImageOptions{
+			Repository: authCfg.ServerAddress + "/" + oc.Namespace() + "/" + repoName,
+			Tag:        tagName,
+		}
+		err = dClient.PullImage(opts, *authCfg)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("removing image...")
+		err = dClient.RemoveImage(opts.Repository)
+		o.Expect(err).NotTo(o.HaveOccurred())
+	})
+})
+
+func waitForImageUpdate(oc *exutil.CLI, image *imageapi.Image) error {
+	return wait.Poll(200*time.Millisecond, 2*time.Minute, func() (bool, error) {
+		newImage, err := oc.AsAdmin().Client().Images().Get(image.Name)
+		if err != nil {
+			return false, err
+		}
+
+		return (image.ResourceVersion < newImage.ResourceVersion), nil
+	})
+}
+
+// deleteTestImagesAndStreams deletes test images built in current and shared
+// namespaces. It also deletes shared projects.
+func deleteTestImagesAndStreams(oc *exutil.CLI) {
+	g.By(fmt.Sprintf("Deleting images and image streams in project %q", oc.Namespace()))
+	iss, err := oc.AdminClient().ImageStreams(oc.Namespace()).List(kapi.ListOptions{})
+	if err != nil {
+		return
+	}
+	for _, is := range iss.Items {
+		for _, history := range is.Status.Tags {
+			for i := range history.Items {
+				oc.AdminClient().Images().Delete(history.Items[i].Image)
+			}
+		}
+		for _, tagRef := range is.Spec.Tags {
```

Comment:
This block seems to be useless since the test doesn't tag the images.

Either remove it or make this a utility function to share it with other tests.